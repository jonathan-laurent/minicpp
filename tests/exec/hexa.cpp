#include <iostream>

int main() {
  std::cout << 0x100 << "\n";
  std::cout << 0xff << "\n";
  std::cout << 0xFF << "\n";
  int maxint = 0x7fffffff;
  std::cout << maxint << "\n";
  int x = 0xDEADBEEF;
  std::cout << x << "\n";
  x = 0xFFFFFFFF;
  std::cout << x << "\n";
  x = 0x80000000;
  std::cout << x << "\n";
}
