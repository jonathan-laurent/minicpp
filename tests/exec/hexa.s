	.text
	jal  _p0__main
	li   $v0, 10
	syscall
_p0__main:
	addi $sp, $sp, -4
	li   $a0, 0x100
	li   $v0, 1
	syscall
	la   $a0, _s__0
	li   $v0, 4
	syscall
	li   $a0, 0xff
	li   $v0, 1
	syscall
	la   $a0, _s__1
	li   $v0, 4
	syscall
	li   $a0, 0xFF
	li   $v0, 1
	syscall
	la   $a0, _s__2
	li   $v0, 4
	syscall
	li   $a0, 0x7fffffff
	li   $v0, 1
	syscall
	la   $a0, _s__3
	li   $v0, 4
	syscall
	li   $a0, 0xDEADBEEF
	li   $v0, 1
	syscall
	la   $a0, _s__4
	li   $v0, 4
	syscall
	li   $a0, 0xFFFFFFFF
	li   $v0, 1
	syscall
	la   $a0, _s__5
	li   $v0, 4
	syscall
	li   $a0, 0x80000000
	li   $v0, 1
	syscall
	la   $a0, _s__6
	li   $v0, 4
	syscall
	addi $sp, $sp, 4
	jr   $ra
	.data
_s__0:
	.asciiz "\n"
_s__1:
	.asciiz "\n"
_s__2:
	.asciiz "\n"
_s__3:
	.asciiz "\n"
_s__4:
	.asciiz "\n"
_s__5:
	.asciiz "\n"
_s__6:
	.asciiz "\n"

