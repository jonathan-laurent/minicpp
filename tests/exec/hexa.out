256
255
255
2147483647
-559038737
-1
-2147483648
